src/controllers/GitHubController.ts

```diff
@@ -0,0 +1,74 @@
+import ProjectData, { ProjectMember } from '../models/ProjectData';
+import * as Octokit from "@octokit/rest";
+
+type RepoType = "public" | "private" | "forks" | "all";
+
+export default class GitHubControler {
+    org: string;
+    type: RepoType;
+    octokit: Octokit;
+
+    constructor(authToken: string, org?: string, type?: RepoType) {
+        this.org = org || "aggie-coding-club";
+        this.type = type || "public";
+        
+        this.octokit = new Octokit({auth: authToken});
+    }
+    
+    getProjectData(): Promise<ProjectData[]> {
+        return this.octokit.repos
+            .listForOrg({
+                org: this.org,
+                type: this.type
+            })
+            .then((reposResponse) => {
+                const repos = reposResponse.data;
+                let projectList: ProjectData[] = [];
+
+                for (let repo of repos) {
+                    let project: ProjectData = {
+                        repoURL: repo.html_url,
+                        name: repo.name,
+                        description: repo.description
+                    };
+                    projectList.push(project);
+                }
+
+                return projectList;
+            }).then(projectList => {
+                let apiCalls = [];
+
+                for (const i in projectList) {
```

Comment:
I actually haven't seen `for...in` used with an iterable before. Ordinarily, I'm all for range-based loops, but this looks really close to `for...of`, which makes the first few reads of this a bit confusing. I get what you're going for here, I'm just not sure it's the most readable approach as compared to a `for (const i = 0...)` approach, because it relies on some in-depth JS knowledge. This is more of a nitpick than anything else, though.

In case you want to switch to a traditional `for` loop syntax, I'll add the suggestion below. If you want to keep this, that's cool too.

```suggestion
                for (let i = 0; i < projectList.length; ++i) {
```